homeassistant/components/kostal_plenticore/config_flow.py

```diff
@@ -0,0 +1,81 @@
+"""Config flow for Kostal Plenticore Solar Inverter integration."""
+import asyncio
+import logging
+
+from aiohttp.client_exceptions import ClientError
+from kostal.plenticore import PlenticoreApiClient, PlenticoreAuthenticationException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_BASE, CONF_HOST, CONF_NAME, CONF_PASSWORD
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default="Plenticore"): str,
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+@callback
+def configured_instances(hass):
+    """Return a set of configured Kostal Plenticore HOSTS."""
+    return {
+        entry.data[CONF_HOST] for entry in hass.config_entries.async_entries(DOMAIN)
+    }
+
+
+async def test_connection(hass: HomeAssistant, data) -> None:
+    """Test the connection to the inverter.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+
+    session = async_get_clientsession(hass)
+    async with PlenticoreApiClient(session, data["host"]) as client:
+        await client.login(data["password"])
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Kostal Plenticore Solar Inverter."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize a new ConfigFlow instance."""
+        self._errors = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        self._errors = {}
+
+        if user_input is not None:
+            if user_input[CONF_HOST] not in configured_instances(self.hass):
+                try:
+                    await test_connection(self.hass, user_input)
+
+                    return self.async_create_entry(
+                        title=user_input[CONF_NAME], data=user_input
```

Comment:
Use the host as entry title.

Reply:
I changed this to use the read hostname of the inverter, because if I use the host and the host is an IP the entity gets strange looking names like `sensor.192.168.1.1_home_power`.